From 92ec722018afd8185874faa4c6987765a849701d Mon Sep 17 00:00:00 2001
From: allen-bst <allen@bluestacks.com>
Date: Tue, 26 Sep 2023 16:49:47 +0800
Subject: [PATCH 022/224] [A13] Disable quota limit irregard of fuse and
 sdcardfs

When recording a video using the system camera app, we encountered the  error log below:
"StorageManager: Failed to set extended attributes for /storage/emulated/0/xxx ..."
As a result, we cannot save the video. To fix this issue, we have disabled the quota limit.
---
 .../jni/android_os_storage_StorageManager.cpp | 37 -------------------
 1 file changed, 37 deletions(-)

diff --git a/core/jni/android_os_storage_StorageManager.cpp b/core/jni/android_os_storage_StorageManager.cpp
index 3116cc8f19e6..7caac72c9bed 100644
--- a/core/jni/android_os_storage_StorageManager.cpp
+++ b/core/jni/android_os_storage_StorageManager.cpp
@@ -30,43 +30,6 @@ namespace android {
 
 jboolean android_os_storage_StorageManager_setQuotaProjectId(JNIEnv* env, jobject self,
                                                              jstring path, jlong projectId) {
-    struct fsxattr fsx;
-    ScopedUtfChars utf_chars_path(env, path);
-
-    static bool sdcardFsSupported = IsSdcardfsUsed();
-    if (sdcardFsSupported) {
-        // sdcardfs doesn't support project ID quota tracking and takes care of quota
-        // in a different way.
-        return JNI_TRUE;
-    }
-
-    if (projectId > UINT32_MAX) {
-        LOG(ERROR) << "Invalid project id: " << projectId;
-        return JNI_FALSE;
-    }
-
-    android::base::unique_fd fd(
-            TEMP_FAILURE_RETRY(open(utf_chars_path.c_str(), O_RDONLY | O_CLOEXEC)));
-    if (fd == -1) {
-        PLOG(ERROR) << "Failed to open " << utf_chars_path.c_str() << " to set project id.";
-        return JNI_FALSE;
-    }
-
-    int ret = ioctl(fd, FS_IOC_FSGETXATTR, &fsx);
-    if (ret == -1) {
-        PLOG(ERROR) << "Failed to get extended attributes for " << utf_chars_path.c_str()
-                    << " to get project id.";
-        return JNI_FALSE;
-    }
-
-    fsx.fsx_projid = projectId;
-    ret = ioctl(fd, FS_IOC_FSSETXATTR, &fsx);
-    if (ret == -1) {
-        PLOG(ERROR) << "Failed to set extended attributes for " << utf_chars_path.c_str()
-                    << " to set project id.";
-        return JNI_FALSE;
-    }
-
     return JNI_TRUE;
 }
 
